scipy/stats/_kde.py

```diff
@@ -260,7 +260,7 @@ def evaluate(self, points):
                 points = reshape(points, (self.d, 1))
                 m = 1
             else:
-                msg = "points have dimension %s, dataset has dimension %s" % (d,
+                msg = "points have dimension {}, dataset has dimension {}".format(d,
                     self.d)
```

Comment:
Was there a reason for not converting this to an f-string, like most of the other formatted string changes?

Reply:
Just that those changes were made by ruff's equivalent of pyupgrade, I didn't touch much by hand. Happy to upgrade them.

Reply:
> Happy to upgrade them.

Not necessary.  I see that it happened elsewhere too.  That kind of refinement can be done in a follow-up, if there is agreement on continuing these types of stylistic improvement.